tarantool/msgpack_ext/types/datetime.py

```diff
@@ -0,0 +1,260 @@
+    else:
+        return datetime, ''
+
+class Datetime():
+    def __init__(self, data=None, *, timestamp=None, year=None, month=None,
+                 day=None, hour=None, minute=None, sec=None, nsec=None,
+                 tzoffset=0, tz=''):
+        if data is not None:
+            if not isinstance(data, bytes):
+                raise ValueError('data argument (first positional argument) ' +
+                                 'expected to be a "bytes" instance')
+
+            datetime, tz = msgpack_decode(data)
+            self._datetime = datetime
+            self._tz = tz
+            return
+
+        # The logic is same as in Tarantool, refer to datetime API.
+        # https://www.tarantool.io/en/doc/latest/reference/reference_lua/datetime/new/
+        if timestamp is not None:
+            if ((year is not None) or (month is not None) or \
+                    (day is not None) or (hour is not None) or \
+                    (minute is not None) or (sec is not None)):
+                raise ValueError('Cannot provide both timestamp and year, month, ' +
+                                 'day, hour, minute, sec')
+
+            if nsec is not None:
+                if not isinstance(timestamp, int):
+                    raise ValueError('timestamp must be int if nsec provided')
+
+                total_nsec = timestamp * NSEC_IN_SEC + nsec
+                datetime = pandas.to_datetime(total_nsec, unit='ns')
+            else:
+                datetime = pandas.to_datetime(timestamp, unit='s')
+        else:
+            if nsec is not None:
+                microsecond = nsec // NSEC_IN_MKSEC
+                nanosecond = nsec % NSEC_IN_MKSEC
+            else:
+                microsecond = 0
+                nanosecond = 0
+
+            datetime = pandas.Timestamp(year=year, month=month, day=day,
+                                        hour=hour, minute=minute, second=sec,
+                                        microsecond=microsecond,
+                                        nanosecond=nanosecond)
+
+        if tz != '':
+            if tz not in tt_timezones.timezoneToIndex:
+                raise ValueError(f'Unknown Tarantool timezone "{tz}"')
+
+            tzinfo = get_python_tzinfo(tz, ValueError)
+            self._datetime = datetime.replace(tzinfo=tzinfo)
+            self._tz = tz
+        elif tzoffset != 0:
+            tzinfo = pytz.FixedOffset(tzoffset)
+            self._datetime = datetime.replace(tzinfo=tzinfo)
+            self._tz = ''
+        else:
+            self._datetime = datetime
+            self._tz = ''
+
+    def __eq__(self, other):
+        if isinstance(other, Datetime):
+            return self._datetime == other._datetime
+        elif isinstance(other, pandas.Timestamp):
+            return self._datetime == other
+        else:
+            return False
+
+    def __str__(self):
+        return self._datetime.__str__()
```

Comment:
pandas.Timestamp string representation do not display timezone name at all.
```python
>>> dt = tarantool.Datetime(year=1970, month=1, day=1)
>>> str(dt)
'1970-01-01 00:00:00'
>>> dt = tarantool.Datetime(year=1970, month=1, day=1, tz='MSK')
>>> str(dt)
'1970-01-01 00:00:00+03:00'
>>> dt = tarantool.Datetime(year=1970, month=1, day=1, tz='Europe/Moscow')
>>> str(dt)
'1970-01-01 00:00:00+03:00'
```

In case of `__repr__`, we display honest internals that could be used to build the same Datetime object (as expected of `__repr__`)
```
>>> tarantool.Datetime(year=1970, month=1, day=1)
datetime: Timestamp('1970-01-01 00:00:00'), tz: ""
>>> tarantool.Datetime(year=1970, month=1, day=1, tz='MSK')
datetime: Timestamp('1970-01-01 00:00:00+0300', tz='pytz.FixedOffset(180)'), tz: "MSK"
>>> tarantool.Datetime(year=1970, month=1, day=1, tz='Europe/Moscow')
datetime: Timestamp('1970-01-01 00:00:00+0300', tz='Europe/Moscow'), tz: "Europe/Moscow"
```